internal/clientcache/internal/db/db.go

```diff
@@ -50,16 +57,38 @@ func Open(ctx context.Context, opt ...Option) (*db.DB, error) {
 	conn.Debug(opts.withDebug)
 
 	switch {
-	case opts.withDbType == dbw.Sqlite:
+	case opts.withDbType == dbw.Sqlite && url == DefaultStoreUrl:
```

Comment:
Yes, since there are a ton of tests that rely on this (don't setup a dir/filename).  There's only one place we need to specify the file based url and it's clear in the code.